migrate.sh

```diff
@@ -24,89 +32,126 @@ verify_run() {
-  for file in .circleci/*; do
+# Gets all the YAML files in the src/<component> directory
+get_components() {
+  # Return an array of command names as given by the filename in `src/<component>/`
+  local -a component_names
+  component_dir="./src/${1}"
+  for file in "${component_dir}"/*.yaml "${component_dir}"/*.yml; do
     if [ -f "$file" ]; then
-      mv "$file" "$file.bak"
+      component_names+=("$file")
     fi
   done
+  echo "${component_names[@]}"
 }
 
-download_template() {
-  ORB_TEMPLATE_TAG_LIST=$(curl --request GET \
-    --url https://api.github.com/repos/CircleCI-Public/Orb-Template/tags \
-    --header 'Accept: application/vnd.github.v3+json' | jq -r '.[].name' | grep -v '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -Vr)
-  ORB_TEMPLATE_VERSION=$(echo "$ORB_TEMPLATE_TAG_LIST" | head -n 1)
-  ORB_TEMPLATE_DOWNLOAD_URL="https://github.com/CircleCI-Public/Orb-Template/archive/refs/tags/${ORB_TEMPLATE_VERSION}.tar.gz"
-  ORB_TEMP_DIR=$(mktemp -d)
-
-  curl -Ls "$ORB_TEMPLATE_DOWNLOAD_URL" -o "$ORB_TEMP_DIR/orb-template.tar.gz"
-  tar -xzf "$ORB_TEMP_DIR/orb-template.tar.gz" -C "$ORB_TEMP_DIR" --strip-components 1
-  cp -r "${ORB_TEMP_DIR}/.circleci/." .circleci/
+convert_to_snake_case() {
+  # Convert the string to snake_case
+  echo "$1" | sed -E 's/-/_/g' | tr '[:upper:]' '[:lower:]'
+}
 
-  if [[ ! -e .yamllint ]]; then
-    cp "${ORB_TEMP_DIR}/.yamllint" .yamllint
-  fi
+# Takes in a list of files and renames them to snake_case and standardizes the file extension to .yml
+rename_components() {
+  for file in ${1}; do
+    # Get the filename without the extension
+    filename=$(basename -- "$file")
+    filename="${filename%.*}"
+    token_map+=("$filename")
+    # Convert the filename to snake_case
+    filename=$(convert_to_snake_case "$filename")
+    filepath=$(dirname "$file")
+    # Rename the file
+    mv "$file" "${filepath}/${filename}.yml" 2>/dev/null
+  done
 }
 
-copy_custom_components() {
-  ORIGINAL_EXECUTORS=$(yq '.executors' .circleci/config.yml.bak)
-  ORIGINAL_JOBS=$(yq '.jobs' .circleci/config.yml.bak)
-  ORIGINAL_COMMANDS=$(yq '.commands' .circleci/config.yml.bak)
-  export ORIGINAL_EXECUTORS
-  export ORIGINAL_JOBS
-  export ORIGINAL_COMMANDS
-  if [[ -n "$ORIGINAL_EXECUTORS" && ! "$ORIGINAL_EXECUTORS" == "null" ]]; then
-    yq -i '. += {"executors": env(ORIGINAL_EXECUTORS)}' .circleci/test-deploy.yml
-  fi
-  if [[ -n "$ORIGINAL_JOBS" && ! "$ORIGINAL_JOBS" == "null" ]]; then
-    yq -i '. += {"jobs": env(ORIGINAL_JOBS)}' .circleci/test-deploy.yml
-  fi
-  if [[ -n "$ORIGINAL_COMMANDS" && ! "$ORIGINAL_COMMANDS" == "null" ]]; then
-    yq -i '. += {"commands": env(ORIGINAL_COMMANDS)}' .circleci/test-deploy.yml
-  fi
+destroy_script() {
+  rm -f "$0"
 }
 
-user_input() {
-  read -rp 'Namespace: ' ORB_NAMESPACE
-  read -rp 'Orb name: ' ORB_NAME
-  read -rp 'Context name: ' ORB_CONTEXT_NAME
+find_and_replace() {
+  #1 the file
+  #2 the string to find
+  #3 the string to replace it with
+  sed -i "s/$2/$3/g" "$1"
```

Comment:
To make it compatible with macOS:

```suggestion
  sed -i "" -e "s/$2/$3/g" "$1"
```